contracts/libs/0.8.x/BytecodeStorageV1.sol

```diff
@@ -303,9 +338,10 @@ library BytecodeStorage {
     ) private view returns (bytes32 version) {
         // get the size of the data
         uint256 bytecodeSize = _bytecodeSizeAt(_address);
-        // handle case where address contains code < minimum expected version string size
+        // handle case where address contains code < minimum expected version string size,
+        // by returning early with the unknown version string
         if (bytecodeSize < (VERSION_OFFSET + 32)) {
-            revert("ContractAsStorage: Read Error");
+            return UNKNOWN_VERSION_STRING;
```

Comment:
nit: since this is a private function, and we no longer perform any actions other than reverting when we return `UNKNOWN_VERSION_STRING`, I think we could just revert here (and in the assembly block below) instead of returning the unknown version string.

Additionally, we could return an [enum](https://docs.soliditylang.org/en/latest/types.html#enums) (e.g. `enum SupportedBytecodeStorageVersions { V0, V1 }`) instead of the version strings, which would help bound the possible return values and clean up some of the checks in `_bytecodeAddressOffsetAt` and `_bytecodeDataOffsetAt`.

If you think the current implementation is easier to grok, I would be fine leaving as-is (no super strong opinions from me here)

Reply:
This function is private, but `getLibraryVersionForBytecode` is internal (and will be public in the new split library) so I think we do want to keep a return value for the "unknown" state.

Given that I anticipate we may have minor versions over time, I am generally inclined to keep the versioning to be a `bytes32` string for now (note: us bounding this to a `bytes32` slot rather than allowing it to be an unbounded `string` is an important consideration in my rationale for keeping it this way, I'd feel differently about using `string) – and consider adding an "enum wrapper" down the road if we think that would be clearer to manage. 

I don't feel strongly about this though, so happy to adjust in an iteration if you feel more strongly than initially surfaced here or if we get feedback from prospective library clients who would like us to expose this functionality.

For now, going to resolve this and merge this PR down into #670, but can do additional iteration in future PRs as needed.

Reply:
Got it and everything you notes makes sense to me! Thanks for double clicking here 🙏

Reply:
> Got it and everything you notes makes sense to me! Thanks for double clicking here 🙏

Vibey – TY ser 💜